### ISSUE TITLE
The Datasets for fine tuning Language Model

### ISSUE THREAD

**Issue by lingsond** [deepset-ai/FARM#355]
Hallo,

I have questions regarding fine tuning a language model. I want to fine tune the bert-base-german-cased pretrained model with old german texts.

I've follow the steps in the lm_finetuning.py script from examples. And got a problem in step 3 (Create a DataSilo), where from the comment on step 3 and the error message:
No such file or directory: '../datasets/german/train.txt'
it expected a train.txt, dev.txt, and also a test.txt file to exist in the data directory (here: ..datasets/german/).

In the example, it just read from a data directory, but it didn't describe anything further. From my understanding/assumption, the data directory would contain the text files (in this case the old German texts, and to make it simple, let's say there are 10 of them and they are named 1.txt, 2.txt, and so on) that are to be used for the fine tuning.

So, to my questions:
- Did I misunderstand the purpose/usage of fine tuning language models?
- If I didn't misunderstand it, then what are supposed to be the content of train.txt, dev.txt, test.txt? Should I join all the text files (1.txt to 10.txt) into one train.txt file? If so, then still the question remains, what should I put in dev.txt and test.txt?

Thanks in advance for the help.

**Comment by lingsond** [comment]
I'm sorry, I'm quite a newbie with github. I didn't notice that the showed issue are the open issues so I didn't look into the closed issues.

I've looked into the closed issues and saw something similar to my question. It has answered my first question, and also how I should structure the train.txt file.

However, I'm still not clear what should be put in the dev.txt and test.txt file.

Again, sorry for before. And thanks in advance for helping.

**Comment by Timoeller** [comment]
Hey, no problem, good that you found part of the solution yourself already! : )

You can either specify dev_filename = None and test_filename = None in the BertStyleLMProcessor, so it wont require those files.
Or you split away small parts of train.txt to test and dev.

On dev and test files the Masked language Model loss and Next Sentence Prediction loss are computed to give a rough indication how finetuning is going.

**Comment by lingsond** [comment]
Thanks for the quick answer. It helps to solve the problem.
